bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/AbstractMQTTThingHandler.java

```diff
@@ -262,4 +304,73 @@ public void postChannelCommand(ChannelUID channelUID, Command command) {
     public void setConnection(MqttBrokerConnection connection) {
         this.connection = connection;
     }
+
+    @Override
+    public void addAvailabilityTopic(String availability_topic, String payload_available,
+            String payload_not_available) {
+        availabilityStates.computeIfAbsent(availability_topic, topic -> {
+            Value value = new OnOffValue(payload_available, payload_not_available);
+            ChannelGroupUID groupUID = new ChannelGroupUID(getThing().getUID(), "availablility");
+            ChannelUID channelUID = new ChannelUID(groupUID, UIDUtils.encode(topic));
+            ChannelState state = new ChannelState(ChannelConfigBuilder.create().withStateTopic(topic).build(),
+                    channelUID, value, new ChannelStateUpdateListener() {
+                        @Override
+                        public void updateChannelState(ChannelUID channelUID, State value) {
+                            calculateThingStatus();
+                        }
+
+                        @Override
+                        public void triggerChannel(ChannelUID channelUID, String eventPayload) {
+                        }
+
+                        @Override
+                        public void postChannelCommand(ChannelUID channelUID, Command value) {
+                        }
+                    });
+            MqttBrokerConnection connection = getConnection();
+            if (connection != null) {
+                state.start(connection, scheduler, 0);
+            }
+
+            return state;
+        });
+    }
+
+    @Override
+    public void removeAvailabilityTopic(@NonNull String availability_topic) {
+        availabilityStates.computeIfPresent(availability_topic, (topic, state) -> {
+            if (connection != null && state != null) {
+                state.stop();
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public void clearAllAvailabilityTopics() {
+        Set<String> topics = new HashSet<>(availabilityStates.keySet());
+        topics.forEach(t -> removeAvailabilityTopic(t));
+    }
+
+    @Override
+    public void resetMessageReceived() {
+        if (messageReceived.getAndSet(false)) {
+            calculateThingStatus();
+        }
+    }
+
+    protected void calculateThingStatus() {
+        final boolean availabilityTopicsSeen;
+
+        if (availabilityStates.isEmpty()) {
+            availabilityTopicsSeen = true;
+        } else {
+            availabilityTopicsSeen = availabilityStates.values().stream().allMatch(
+                    c -> c != null && OnOffType.ON.equals(c.getCache().getChannelState().as(OnOffType.class)));
+        }
+        logger.debug("thing status {} {} {}", getThing().getUID(), messageReceived.get(), availabilityTopicsSeen);
```

Comment:
If the correct logic behaviour is covered by tests, I strongly recommend not to log anything. `updateThingStatus` will already notify the framework and log the new thing status, why add log pressure here as well.

Reply:
just that this `updateThingStatus` is not calling the framwork directly, but an abstract method, which determins the status. I might ignore one or the other parameter.
I was for me, well, a debuging help....
But I can remove it...